src/mail-app/mail/view/MailViewModel.ts

```diff
@@ -230,16 +240,26 @@ export class MailViewModel {
 	}
 
 	private async loadExplicitMailTarget(listId: Id, mailId: Id, onMissingTargetEmail: () => unknown) {
+		const expectedStickyMailId: IdTuple = [listId, mailId]
+
+		// shouldStop returning true guarantees we won't load anything if the mail is not in the list already, so
+		// this will be synchronous
+		const listMail = await this.listModel?.loadAndSelect(mailId, () => true)
+		if (listMail) {
+			console.log(TAG, "opening mail from list", mailId)
+			return
+		}
+
 		const cached = await this.cacheStorage.get(MailTypeRef, listId, mailId)
 		if (cached) {
 			console.log(TAG, "opening cached mail", mailId)
+			await this.resetOrInitializeList([listId, mailId])
```

Comment:
I am confused, above we call `loadAndSelect()` and this function is supposed to deselect?

Reply:
Let's say we already have a correct folder loaded and then we click on notification. Let's say the email is already in the list. Wouldn't we `loadAndSelect()` to select it and then `resetOrInitializeList()` which will deselect it?

Reply:
Ahhhh yes sorry!